client/lib/gsuite/index.js

```diff
@@ -95,6 +95,17 @@ function getEligibleGSuiteDomain( selectedDomainName, domains ) {
 	return get( primaryDomain, 'name', '' );
 }
 
+/**
+ * Retrieves the first domain name that is eligible to G Suite
+ *
+ * @param {Array} domains - list of domain objects
+ * @returns {String} - Eligible domain name
+ */
+function getFirstEligibleGSuiteDomainName( domains ) {
+	const supportedDomains = getGSuiteSupportedDomains( domains );
+	return supportedDomains.length ? supportedDomains[ 0 ].name : null;
+}
```

Comment:
I'm not convinced that we should introduce another function in addition to `getEligibleGSuiteDomain()`. My thinking is that we could update the latter to return an eligible domain, with a preference for the primary domain. Could you give it a try?

Reply:
Sure. My initial reservation to refactor that method was because there may be some components who rely on that method's behaviour as-is, but your suggestion makes sense.